packages/astro/src/integrations/index.ts

```diff
@@ -30,6 +31,22 @@ export async function runHookConfigDone({ config }: { config: AstroConfig }) {
 		if (integration.hooks['astro:config:done']) {
 			await integration.hooks['astro:config:done']({
 				config,
+				setAdapter(adapter) {
+					config._ctx.adapter = adapter;
+				}
```

Comment:
Can we throw if `config._ctx.adapter` is already set?

Reply:
I added this.